processor/batchprocessor/batch_processor.go

```diff
@@ -241,8 +270,21 @@ func (b *shard) startLoop() {
 	}
 }
 
-func (b *shard) processItem(item any) {
-	b.batch.add(item)
+func (b *shard) processItem(item dataItem) {
+	before := b.batch.itemCount()
+	b.batch.add(item.data)
+	after := b.batch.itemCount()
```

Comment:
Done. Also, since you like simplification, a little more: https://github.com/open-telemetry/opentelemetry-collector/pull/11324/commits/b12c78ae557f3048020d6ddd19261f631189b21f